Sources/Classes/core/Storage/MiniAppSecureStorage.swift

```diff
@@ -90,29 +91,53 @@ public class MiniAppSecureStorage: MiniAppSecureStorageDelegate {
         return storage[key]
     }
 
+    // swiftlint:disable function_body_length
     public func set(dict: [String: String], completion: ((Result<Bool, MiniAppSecureStorageError>) -> Void)? = nil) {
-        guard storageFileSize <= fileSizeLimit else {
-            completion?(.failure(MiniAppSecureStorageError.storageFullError))
+        guard let memorySize = try? getMemoryStorageFileSize() else {
+            completion?(.failure(.storageUnvailable))
+            return
+        }
+        guard memorySize <= fileSizeLimit else {
+            completion?(.failure(.storageFullError))
             return
         }
         guard storage != nil else {
-            completion?(.failure(MiniAppSecureStorageError.storageUnvailable))
+            completion?(.failure(.storageUnvailable))
             return
         }
         guard !isBusy else {
-            completion?(.failure(MiniAppSecureStorageError.storageBusy))
+            completion?(.failure(.storageBusy))
             return
         }
         isBusy = true
         for (key, value) in dict {
-            MiniAppLogger.d("🔑 Secure Storage: set '\(key)'")
+            MiniAppLogger.d("🔑 Secure Storage: will set '\(key)'")
+            let estimateAddString = "<key>" + key + "</key><string>" + value + "</string>"
+            guard
+                let estimateAddData = estimateAddString.data(using: .utf8),
+                let memorySize = try? getMemoryStorageFileSize()
+            else { completion?(.failure(.storageIOError))
+                return
+            }
+            let estimatedAddSize = UInt64(estimateAddData.count)
+            let estimatedFinalSize = memorySize + estimatedAddSize
+            guard estimatedFinalSize <= fileSizeLimit else {
+                completion?(.failure(.storageFullError))
+                return
+            }
```

Comment:
So we are making sure that the limit is not exceeded before making any new entry into the storage file. 🤔 
Can we move this to a seperate method?

Reply:
I've updated it with a slight improvement so I'm checking if the whole dict is sufficient to fit into the available storage instead for every key + extraction